.github/workflows/ci.yml

```diff
@@ -36,6 +36,17 @@ jobs:
             target
           key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
 
+      - name: build
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
```

Comment:
I think test will automatically build. we probably don't need this step. What do you think?

Reply:
Ah right, thanks for noticing this. We can remove the build step since test step will cover it.

However, I added build in case there are tests that do not cover some part of code, in which case it will not compile everything and thus won't throw error for that part (if there are any issues). I see that there is ~82.5% coverage but just wanted to confirm some component is not being tested fully before I go ahead and remove it.